deepchecks/checks/distribution/whole_dataset_drift.py

```diff
@@ -0,0 +1,234 @@
+        """
+
+        train_dataset = Dataset.validate_dataset_or_dataframe(train_dataset)
+        test_dataset = Dataset.validate_dataset_or_dataframe(test_dataset)
+
+        features = train_dataset.validate_shared_features(test_dataset, self.__class__.__name__)
+        cat_features = train_dataset.validate_shared_categorical_features(test_dataset, self.__class__.__name__)
+        self._cat_features = cat_features
+
+        model = self._generate_model(list(set(features) - set(cat_features)), cat_features)
+
+        sample_size = min(self.sample_size, train_dataset.n_samples, test_dataset.n_samples)
+        train_sample_df = train_dataset.features_columns.sample(sample_size, random_state=self.random_state)
+        test_sample_df = test_dataset.features_columns.sample(sample_size, random_state=self.random_state)
+
+        # create new dataset, with label denoting whether sample belongs to test dataset
+        domain_class_df = pd.concat([train_sample_df, test_sample_df])
+        domain_class_labels = pd.Series([0] * len(train_sample_df) + [1] * len(test_sample_df))
+
+        from sklearn.model_selection import train_test_split
+        x_train, x_test, y_train, y_test = train_test_split(domain_class_df, domain_class_labels,
+                                                            stratify=domain_class_labels,
+                                                            random_state=self.random_state,
+                                                            test_size=0.3)
+
+        model = model.fit(x_train, y_train)
+
+        y_test.name = 'belongs_to_test'
+        domain_test_dataset = Dataset(pd.concat([x_test.reset_index(drop=True), y_test.reset_index(drop=True)], axis=1),
+                                      cat_features=cat_features, label='belongs_to_test')
+
+        # calculate feature importance of model, containing the information of what features can separate the datasets
+        fi_ser = calculate_feature_importance(model, domain_test_dataset,
+                                              force_permutation=True,
+                                              permutation_wkargs={'n_repeats': 10}).sort_values(ascending=False)
+
+        values_dict = {
+            'domain_classifier_auc': roc_auc_score(y_test, model.predict_proba(x_test)[:, 1]),
+            'domain_classifier_feature_importance': fi_ser.to_dict(),
+        }
+
+        headnote = f"""<span>
+                    The shown features are the features that are most important for the domain classifier - the model
+                    trained to distinguish between the train and test datasets.<br> The percents of explained dataset 
+                    difference are the calculated feature importance values for the feature.
+                </span>"""
+
+        top_fi = fi_ser.head(self.n_top_features)
+        top_fi = top_fi.loc[top_fi > self.min_feature_importance]
```

Comment:
should be one or the other IMO, not both

Reply:
IMO both is fine, I do want to hard limit to a certain number, and also cut off unseals features.

Reply:
What happens if no features pass this minimum?

Reply:
Disagree but letting go